GVFS/GVFS.Common/Maintenance/MaintenanceJobManager.cs

```diff
@@ -0,0 +1,341 @@
+﻿using GVFS.Common.FileSystem;
+using GVFS.Common.Git;
+using GVFS.Common.Prefetch;
+using GVFS.Common.Tracing;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+
+namespace GVFS.Common.Maintenance
+{
+    public class MaintenanceJobManager : IDisposable

```

Comment:
As I noted in your design doc, I think this name is too generic. I would rename it to reference "CacheMaintenance" so we can later on extend to repo maintenance as well.